data/scripting/infrastructure.lua

```diff
@@ -65,8 +65,8 @@ function connected_road(roadtype, p, start, cmd, g_create_carriers)
          r = p:place_road(roadtype, start, table.unpack(moves))
          start = r.end_flag
          if create_carriers then
-            if roadtype == "normal" then r:set_workers(p.tribe.carrier, 1)
-            elseif roadtype == "busy" then r:set_workers({[p.tribe.carrier] = 1, [p.tribe.carrier2] = 1})
+            if roadtype == "normal" then r:set_workers(p.tribe.carriers[1], 1)
+            elseif roadtype == "busy" then r:set_workers({[p.tribe.carriers[1]] = 1, [p.tribe.carriers[2]] = 1})
```

Comment:
Actually I wonder if what is desired here is not to place *one of each* type of carrier on the busy road. This is only really relevant for scripts for the Europeans add-on currently but this probably shouldn't be hard to fix while you're already modernizing this line.

Reply:
@SimonHeimberg the 2 lines could be made one line using the valid_workers property of the road.

Reply:
Done. Almost at least. Does the ferry need different handling?

Reply:
@Noordfrees looks good to me valid_workers should include ferries.